src/main/java/org/jabref/logic/util/io/FileUtil.java

```diff
@@ -84,6 +82,28 @@ public static String getFileName(String fileNameWithExtension) {
     }
 
     /**
+     * Returns a valid filename for most operating systems.
+     *
+     * Currently, only the length is restricted to 255 chars.
+     */
+    public static String getValidFileName(String fileName) {
+        String nameWithoutExtension = getFileName(fileName);
+
+        if (nameWithoutExtension.length() > 255) {
```

Comment:
I would suggest to extract this magic number into `public static final int MAX_FILENAME_LENGTH`.  This can then be reused by the test as well.